src/factory/src/api.rs

```diff
@@ -68,8 +71,9 @@ impl TokenFactoryCanister {
 
     #[init]
     pub fn init(&self, controller: Principal, ledger_principal: Option<Principal>) {
-        let ledger = ledger_principal
-            .unwrap_or_else(|| Principal::from_text(DEFAULT_LEDGER_PRINCIPAL).unwrap());
+        let ledger = ledger_principal.unwrap_or_else(|| {
+            Principal::from_text(DEFAULT_LEDGER_PRINCIPAL).expect("const value conversion")
```

Comment:
Can we use the actual `const` function here instead? `Principal::from_slice` I think it is